app/kubernetes/services/kubeconfigService.js

```diff
@@ -0,0 +1,21 @@
+import angular from 'angular';
+
+class KubernetesConfigService {
+  /* @ngInject */
+  constructor(KubernetesConfig, FileSaver) {
+    this.KubernetesConfig = KubernetesConfig;
+    this.FileSaver = FileSaver;
+  }
+
+  async downloadConfig() {
+    try {
+      const response = await this.KubernetesConfig.get();
+      return this.FileSaver.saveAs(response.data, 'config.yaml');
+    } catch (err) {
```

Comment:
catching and throwing the same error is redundant. you can just remove the `try catch`